src/components/Subnav/sgds-subnav.ts

```diff
@@ -0,0 +1,267 @@
+import SgdsElement from "../../base/sgds-element";
+import { html, nothing } from "lit";
+import { query, state } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+import { watch } from "../../utils/watch";
+import { waitForEvent } from "../../utils/event";
+import { animateTo, shimKeyframesHeightAuto, stopAnimations } from "../../utils/animate";
+import { getAnimation, setDefaultAnimation } from "../../utils/animation-registry";
+import { LG_BREAKPOINT } from "../../utils/breakpoints";
+import SgdsIcon from "../Icon/sgds-icon";
+import subnavStyle from "./subnav.css";
+import gridStyle from "../../css/grid.css";
+
+/**
+ * @summary This component provides secondary navigation within a specific section or page. It typically appears below the main navigation and offers context-specific links or actions to help users explore related content.
+ *
+ * @slot default - Default slot of SgdsSubnav. Pass in SgdsSubnavItem elements here.
+ * @slot header - Slot for rendering the sub-navigation header or section title.
+ * @slot actions - Slot for inserting contextual action elements such as buttons, filters, or other controls aligned with the sub-navigation.
+ *
+ */
+
+export class SgdsSubnav extends SgdsElement {
+  static styles = [...SgdsElement.styles, subnavStyle, gridStyle];
+  /** @internal */
+  static dependencies = {
+    "sgds-icon": SgdsIcon
+  };
+
+  /** @internal */
+  @query("nav") nav: HTMLElement;
+
+  /** @internal */
+  @query(".subnav") subnav: HTMLElement;
+
+  /** @internal */
+  @query(".subnav-nav-mobile") mobileNav: HTMLElement;
+
```

Comment:
i think if u move to the next line can add the private than no need the `/**@internal **/`
```
@query()
private mobileNav
```